app/services/duration_parser.rb

```diff
@@ -0,0 +1,39 @@
+# Parses duration strings ("1d", 2w", "3m", "4y" into ActiveSupport::Durations)
+class DurationParser
+  attr_reader :value
+
+  # @param value [String, nil]
+  def initialize(value)
+    @value = value
+  end
+
+  # @return [ActiveSupport::Duration, nil]
+  def parse
+    return if value.blank?
+
+    match = value.match(/^(?<number>\d+)(?<duration>\D)$/)
+    return nil unless match
+
+    parse_duration(Integer(match[:number], 10), match[:duration])
+  rescue ArgumentError
+    nil
+  end
+
+  def valid?
+    value.blank? || !parse.nil?
+  end
+
+  # @api private
+  def parse_duration(number, duration)
+    case duration
+    when 'd' # days
+      number.days
+    when 'w' # weeks
+      (7 * number).days
+    when 'm' # months
+      (30 * number).days
+    when 'y' # years
+      (365 * number).days
```

Comment:
Are we okay that these aren't strictly accurate? e.g. months with fewer or more than 30 days, or leap years.

Would it work to just...
```suggestion
    when 'w' # weeks
      number.weeks
    when 'm' # months
      number.months
    when 'y' # years
      number.years
```

Reply:
When originally implementing this I decided to make it simpler/dumber so I could easily document it:

See `verified_within` (inside the details dropdown) on https://developers.login.gov/oidc/

The ActiveSupport methods are very nice but they use fractional quantities for some of those so it seemed more predictable this way (so when the fractional quantities add up they don't change hours/minutes as much)

```ruby
1.years.to_i.to_f / 1.day.to_i
=> 365.2425
```

I feel somewhat strongly about leaving it simpler and documented, but if you feel strongly, I could change it here and discuss changing the IDP with the rest of the team

Reply:
> I feel somewhat strongly about leaving it simpler and documented, but if you feel strongly, I could change it here and discuss changing the IDP with the rest of the team

I don't feel strongly.

👍